### PR TITLE
CMP-2940 better data types

### PR DESCRIPTION
**Pull request by nicolas-chaix-didomi**
- Fix return types (`any` into `string`, `Vendor`, `Purpose`)
- Update `UserStatus` interface - must use snake case instead of camel case
- Update comments accordingly
- Add new tests to check `Vendor`, `Purpose`, `UserStatus` object and parameters

### CODE REVIEW

File: testApp/android/app/src/androidTest/java/com/example/reactnativedidomi/UIGettersTest.kt [hunk 1]
```diff
@@ -157,4 +157,18 @@ class UIGettersTest: BaseUITest() {
         assertTextContains(",\"global_consent\":{\"enabled\":[".trim())
         assertTextContains(",\"consent\":{\"enabled\":[".trim())
     }
+
+    @Test
+    fun test_getUserStatus_vendors_globalConsent() {
```
Comment (pmerlet-at-didomi) [inline testApp/android/app/src/androidTest/java/com/example/reactnativedidomi/UIGettersTest.kt:162]: For consistency, method name could be test_getUserStatus_vendors_global_consent

File: testApp/src/Getters.tsx [hunk 2]
```diff
@@ -129,7 +129,17 @@ export default function Getters() {
...
-          return await (await Didomi.getUserStatus()).vendors;
+          return (await Didomi.getUserStatus()).vendors;
+        }}
+        test={() => {
+          return true;
+        }}
+      />
+
+      <Getter
+        name="getUserStatus vendors globalConsent"
```
Comment (pmerlet-at-didomi) [inline testApp/src/Getters.tsx:140]: Same, getter name could be global_consent

File: testApp/android/app/src/androidTest/java/com/example/reactnativedidomi/UIGettersTest.kt (hunk 1 above)
Comment (nicolas-chaix-didomi) [reply testApp/android/app/src/androidTest/java/com/example/reactnativedidomi/UIGettersTest.kt]: change pushed

File: testApp/src/Getters.tsx (hunk 2 above)
Comment (nicolas-chaix-didomi) [reply testApp/src/Getters.tsx]: change pushed

**Review by pmerlet-at-didomi — APPROVED** [review]

**Pull request merged**